packages/contracts/contracts/QuadraticVoteCollectModule.sol

```diff
@@ -0,0 +1,260 @@
+ */
+struct ProfilePublicationData {
+    uint256 amount;
+    address currency;
+    uint96 collectLimit;
+    uint96 currentCollects;
+    address recipient;
+    uint16 referralFee;
+    bool followerOnly;
+    uint72 endTimestamp;
+}
+
+struct VotingData {
+    address grantsRoundAddress;
+    address votingStrategyAddress;
+    uint256 roundStartTime;
+    uint256 roundEndTime;
+}
+
+contract QuadraticVoteCollectModule is FeeModuleBase, FollowValidationModuleBase, ICollectModule, IERC1271, ERC165 {
+    using SafeERC20 for IERC20;
+
+    mapping(uint256 => mapping(uint256 => ProfilePublicationData)) internal _dataByPublicationByProfile;
+
+    /**
+     * @dev Mapping of valid vote signature digests
+     */
+    mapping(bytes32 => bool) internal _validVotes;
+
+    IRoundImplementation public roundImplementation;
+
+    constructor(address _lensHub, address _moduleGlobals) FeeModuleBase(_moduleGlobals) ModuleBase(_lensHub) {}
+
+    function initializePublicationCollectModule(
+        uint256 profileId,
+        uint256 pubId,
+        bytes calldata data
+    ) external returns (bytes memory) {
+        (uint256 amount, address currency, address recipient, uint16 referralFee, bool followerOnly) = abi.decode(
+            data,
+            (uint256, address, address, uint16, bool)
+        );
+        if (!_currencyWhitelisted(currency) || recipient == address(0) || referralFee > BPS_MAX || amount == 0)
+            revert Errors.InitParamsInvalid();
+
+        _dataByPublicationByProfile[profileId][pubId].amount = amount;
+        _dataByPublicationByProfile[profileId][pubId].currency = currency;
+        _dataByPublicationByProfile[profileId][pubId].recipient = recipient;
+        _dataByPublicationByProfile[profileId][pubId].referralFee = referralFee;
+        _dataByPublicationByProfile[profileId][pubId].followerOnly = followerOnly;
+
+        return data;
+    }
+
+    function processCollect(
+        uint256 referrerProfileId,
+        address collector,
+        uint256 profileId,
+        uint256 pubId,
+        bytes calldata data
+    ) external {
+        if (_dataByPublicationByProfile[profileId][pubId].followerOnly) _checkFollowValidity(profileId, collector);
+        if (referrerProfileId == profileId) {
+            _processCollect(collector, profileId, pubId, data);
+        } else {
+            _processCollectWithReferral(referrerProfileId, collector, profileId, pubId, data);
+        }
+    }
+
+    function _processCollect(address collector, uint256 profileId, uint256 pubId, bytes calldata data) internal {
+        //decode data;
+        /// decode  data
+        (address _currency, uint256 _amount, VotingData memory votingData) = abi.decode(
+            data,
+            (address, uint256, VotingData)
+        );
+        require(
+            block.timestamp > votingData.roundStartTime && block.timestamp < votingData.roundEndTime,
```

Comment:
This is risky. Any user can provide data on the round in the encoded data and the checks will pass

Reply:
I changed this to get the time stamps from the round module.  but now I wonder if we should store the current active module in the contract and update it whenever a new round is created.  a malicious actor could create a custom round implementation and pass it in as well.

Reply:
Just let the round handle it. If you're calling the round for the limits, and the round checks on limits, skip the first call

Reply:
Check is in round, removed from module